valhalla: JVM_IsValhallaEnabled is required

From a local build after resolving merging conflict:
```
Undefined symbols for architecture x86_64:
  "_JVM_IsValhallaEnabled", referenced from:
      _Java_jdk_internal_misc_ValhallaFeatures_isValhallaEnabled in ValhallaFeatures.o
```

FYI @hangshao0

There seems to be a merge issue. After adding the method `JVM_IsValhallaEnabled()`, I get:

```
Creating java.base.jmod
Exception in thread "main" java.lang.ExceptionInInitializerError
        at java.base/java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:175)
        at java.base/java.lang.invoke.VarHandle$VarHandleDesc$Kind.<clinit>(VarHandle.java:2305)
        at java.base/java.lang.invoke.VarHandle$1.<clinit>(VarHandle.java:2316)
        at java.base/java.lang.invoke.VarHandle$AccessType.accessModeType(VarHandle.java:1691)
        at java.base/java.lang.invoke.VarHandleByteArrayAsInts$ArrayHandle.accessModeTypeUncached(VarHandleByteArrayAsInts.java:97)
        at java.base/java.lang.invoke.VarHandle.accessModeTypeUncached(VarHandle.java:2115)
        at java.base/java.lang.invoke.VarHandle.accessModeType(VarHandle.java:2109)
        at java.base/sun.security.provider.ByteArrayAccess.b2iBig64(ByteArrayAccess.java:101)
        at java.base/sun.security.provider.SHA2.implCompress0(SHA2.java:145)
        at java.base/sun.security.provider.SHA2.implCompress(SHA2.java:123)
        at java.base/sun.security.provider.DigestBase.engineUpdate(DigestBase.java:123)
        at java.base/java.security.MessageDigest$Delegate.engineUpdate(MessageDigest.java:653)
        at java.base/java.security.MessageDigest.update(MessageDigest.java:347)
        at java.base/jdk.internal.module.ModuleHashes.lambda$computeHash$0(ModuleHashes.java:125)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:523)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:512)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:239)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/jdk.internal.module.ModuleHashes.computeHash(ModuleHashes.java:120)
        at java.base/jdk.internal.module.ModuleHashes.generate(ModuleHashes.java:168)
        at java.base/jdk.internal.module.ModuleHashesBuilder.lambda$computeHashes$3(ModuleHashesBuilder.java:120)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.ArrayDeque$DeqSpliterator.forEachRemaining(ArrayDeque.java:848)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:522)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:512)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:239)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/jdk.internal.module.ModuleHashesBuilder.computeHashes(ModuleHashesBuilder.java:104)
        at jdk.jlink/jdk.tools.jmod.JmodTask$Hasher.computeHashes(JmodTask.java:938)
        at jdk.jlink/jdk.tools.jmod.JmodTask$JmodFileWriter.computeHashes(JmodTask.java:660)
        at jdk.jlink/jdk.tools.jmod.JmodTask$JmodFileWriter.writeModuleInfo(JmodTask.java:591)
        at jdk.jlink/jdk.tools.jmod.JmodTask$JmodFileWriter.write(JmodTask.java:487)
        at jdk.jlink/jdk.tools.jmod.JmodTask.create(JmodTask.java:444)
        at jdk.jlink/jdk.tools.jmod.JmodTask.run(JmodTask.java:198)
        at jdk.jlink/jdk.tools.jmod.Main.main(Main.java:35)
Caused by: java.lang.IllegalArgumentException: not a valid reference type descriptor: Ljava/lang/Object;
        at java.base/java.lang.constant.ClassDescImpl.<init>(ClassDescImpl.java:58)
        at java.base/java.lang.constant.ClassDesc.ofDescriptor(ClassDesc.java:175)
        at java.base/java.lang.constant.ClassDesc.of(ClassDesc.java:81)
        at java.base/java.lang.constant.ConstantDescs.<clinit>(ConstantDescs.java:67)
        ... 41 more
```

`ConstantUtils.skipOverFieldSignature()` returns incorrect value on `Ljava/lang/Object;`. I see there was a recent conflict merge of file `ConstantUtils.java`. @JasonFengJ9

--- a/runtime/j9vm/exports.cmake
+++ b/runtime/j9vm/exports.cmake
@@ -418,3 +418,9 @@ if(J9VM_OPT_JITSERVER)
 		JITServer_CreateServer
 	)
 endif()
+
+if(J9VM_OPT_VALHALLA_VALUE_TYPES)
+	jvm_add_exports(jvm
+		JVM_IsValhallaEnabled
+	)
+endif()

--- a/runtime/j9vm/j9vmnatives.xml
+++ b/runtime/j9vm/j9vmnatives.xml
@@ -318,6 +318,11 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 		<export name="_JVM_CopySwapMemory@44"/>
 		<export name="JVM_BeforeHalt"/>
 		<export name="AsyncGetCallTrace"/>
+
+		<!-- Additions for Valhalla -->
+		<export name="JVM_IsValhallaEnabled">
+			<include-if condition="spec.flags.opt_valhallaValueTypes"/>
+		</export>
 	</exports>
 
 	<exports group="jdk11">

--- a/runtime/j9vm/javanextvmi.cpp
+++ b/runtime/j9vm/javanextvmi.cpp
@@ -445,4 +445,12 @@ JVM_VirtualThreadHideFrames(JNIEnv *env, jobject vthread, jboolean hide)
 }
 #endif /* JAVA_SPEC_VERSION >= 20 */
 
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+JNIEXPORT jboolean JNICALL
+JVM_IsValhallaEnabled()
+{
+	return JNI_TRUE;
+}
+#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
+
 } /* extern "C" */

--- a/runtime/redirector/forwarders.m4
+++ b/runtime/redirector/forwarders.m4
@@ -404,3 +404,5 @@ _IF([JAVA_SPEC_VERSION >= 20],
 	[_X(JVM_GetClassFileVersion, JNICALL, false, jint, JNIEnv *env, jclass cls)])
 _IF([JAVA_SPEC_VERSION >= 20],
 	[_X(JVM_VirtualThreadHideFrames, JNICALL, false, void, JNIEnv *env, jobject vthread, jboolean hide)])
+_IF([defined(J9VM_OPT_VALHALLA_VALUE_TYPES)],
+	[_X(JVM_IsValhallaEnabled, JNICALL, false, jboolean, void)])

